src/gui/properties/peerlistwidget.cpp

```diff
@@ -262,44 +262,53 @@ void PeerListWidget::updatePeerCountryResolutionState()
 
 void PeerListWidget::showPeerListMenu(const QPoint &)
 {
-    BitTorrent::Torrent *const torrent = m_properties->getCurrentTorrent();
+    BitTorrent::Torrent *torrent = m_properties->getCurrentTorrent();
     if (!torrent) return;
 
-    QMenu *menu = new QMenu(this);
+    const auto adjustAction = [](QAction *action, const bool predicate, const QString &tooltip)
+    {
+        if (!predicate)
+            return;
+
+        action->setEnabled(false);
+        action->setToolTip(tooltip);
+    };
+
+    auto *menu = new QMenu(this);
     menu->setAttribute(Qt::WA_DeleteOnClose);
+    menu->setToolTipsVisible(true);
 
     // Add Peer Action
-    // Do not allow user to add peers in a private torrent
-    if (!torrent->isQueued() && !torrent->isChecking() && !torrent->isPrivate())
+    QAction *addNewPeer = menu->addAction(UIThemeManager::instance()->getIcon("user-group-new"), tr("Add a new peer...")
+        , this, [this, torrent]()
     {
-        menu->addAction(UIThemeManager::instance()->getIcon("user-group-new"), tr("Add a new peer...")
-            , this, [this, torrent]()
+        const QVector<BitTorrent::PeerAddress> peersList = PeersAdditionDialog::askForPeers(this);
+        const int peerCount = std::count_if(peersList.cbegin(), peersList.cend(), [torrent](const BitTorrent::PeerAddress &peer)
         {
-            const QVector<BitTorrent::PeerAddress> peersList = PeersAdditionDialog::askForPeers(this);
-            const int peerCount = std::count_if(peersList.cbegin(), peersList.cend(), [torrent](const BitTorrent::PeerAddress &peer)
-            {
-                return torrent->connectPeer(peer);
-            });
-            if (peerCount < peersList.length())
-                QMessageBox::information(this, tr("Adding peers"), tr("Some peers cannot be added. Check the Log for details."));
-            else if (peerCount > 0)
-                QMessageBox::information(this, tr("Adding peers"), tr("Peers are added to this torrent."));
+            return torrent->connectPeer(peer);
         });
-    }
+        if (peerCount < peersList.length())
+            QMessageBox::information(this, tr("Adding peers"), tr("Some peers cannot be added. Check the Log for details."));
+        else if (peerCount > 0)
+            QMessageBox::information(this, tr("Adding peers"), tr("Peers are added to this torrent."));
+    });
+    adjustAction(addNewPeer, torrent->isChecking(), tr("Cannot add peers when torrent is checking"));
```

Comment:
I would prefer to see adjustAction definition as much closer as possible to its first usage.